docs/index.md

```diff
@@ -3,4 +3,51 @@ title: "Introduction"
 weight: 10
 ---
 
->This document is under construction.
+The AsyncAPI Generator is a tool that generates anything you want using the **[AsyncAPI file](asyncapi-file.md)** and **[template](template.md)** that are supplied as inputs to the AsyncAPI Generator CLI. The Generator was built with extensibility in mind; you can use the generator to generate anything you want provided that it can be defined in a template, such as code, diagrams, markdown files, microservices, and applications. A number of [community-maintained templates](https://github.com/search?q=topic%3Aasyncapi+topic%3Agenerator+topic%3Atemplate) are now available for immediate usage.
+
+### Generator use cases 
+- Generation of interactive and understandable API documentation
+- Generation of APIs' client libraries
+- Generation of APIs' boilerplate code
+
+### Generator advantages
+- Quick set up and easy to use on a regular basis
+- Effortless generation of complex documents
+- List of readily available AsyncAPI templates
+- Support for render engines such as React and Nunjucks
+
+### Generation process
+1. The **Generator** receives the **[template](template.md)** and **[AsyncAPI file](asyncapi-file.md)** as inputs. 
+2. The **Generator** sends to the **[Parser](parser.md)** the stringified version of the original specification file in order to validate the JSON/YAML **AsyncAPI file**.
```

Comment:
please adjust all step as I suggested for the other document:
- AsyncAPI file to AsyncAPI document
- use proper cases when referring to items from diagram. So for example if diagram says `Parser` then also write `Parser` in all steps with first capital letter